pandas/tests/series/indexing/test_setitem.py

```diff
@@ -126,6 +128,16 @@ def test_setitem_boolean_different_order(self, string_series):
 
         tm.assert_series_equal(copy, expected)
 
+    def test_setitem_boolean_ea_type(self):
+        # GH: 26468
+        df = DataFrame(
+            {"a": [0, 0, np.nan, np.nan], "b": array(range(4), dtype="Int64")}
+        )
+        s = Series(array([1] * 4, dtype="Int64"))
+        s[df["a"].isna()] = df.loc[df["a"].isna(), "b"]
```

Comment:
can you do this without constructing a DataFrame

Reply:
Avoided the DataFrame construction. Checked, that this raised the same error as reported in the issue.